tests/IntegrationTests/WebAppUiTests/UiTestHelpers.cs

```diff
@@ -66,5 +125,129 @@ await page.Context.Tracing.StopAsync(new()
+            string[] segments = testedAppLocation.Split(Path.DirectorySeparatorChar);
+            int numberSegments = segments.Length;
+            int startLastSegments = numberSegments - 3;
+            int endFirstSegments = startLastSegments - 2;
+            return Path.Combine(
+                Path.Combine(segments.Take(endFirstSegments).ToArray()),
+                appLocation,
+                Path.Combine(segments.Skip(startLastSegments).ToArray())
+            );
+        }
+
+        /// <summary>
+        /// Kills the processes in the queue and all of their children
+        /// </summary>
+        /// <param name="processQueue">queue of parent processes</param>
+        public static void killProcessTrees(Queue<Process> processQueue)
+        {
+            Process currentProcess;
+            while (processQueue.Count > 0)
+            {
+                currentProcess = processQueue.Dequeue();
+                if (currentProcess == null) { continue;}
+
+                foreach (Process child in GetChildProcesses(currentProcess))
+                {
+                    processQueue.Enqueue(child);
+                };
+                currentProcess.Kill();
+                currentProcess.Close();
+            }
+        }
+
+        /// <summary>
+        /// Gets the child processes of a process on Windows
+        /// </summary>
+        /// <param name="process">The parent process</param>
+        /// <returns>A list of child processes</returns>
+        /// <exception cref="NotImplementedException">Thrown if running on an OS other than Windows</exception>
+        public static IList<Process> GetChildProcesses(this Process process)
+        {
+            // Validate platform compatibility
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return new ManagementObjectSearcher(
+                    $"Select * From Win32_Process Where ParentProcessID={process.Id}")
+                    .Get()
+                    .Cast<ManagementObject>()
+#pragma warning disable CA1416 // This call can not be reached except on Windows due to the enclosing if statement
+                    .Select(mo =>
+                        Process.GetProcessById(Convert.ToInt32(mo["ProcessID"], System.Globalization.CultureInfo.InvariantCulture)))
+#pragma warning restore CA1416
+                    .ToList();
+            }
+            else
+            {
+                throw new NotImplementedException("Not implemented for this OS");
+            }
+        }
+
+        /// <summary>
+        /// Checks if all processes in a list are alive
+        /// </summary>
+        /// <param name="processes">List of processes to check</param>
+        /// <returns>True if all are alive else false</returns>
+        public static bool ProcessesAreAlive(List<Process> processes)
+        {
+            foreach (Process process in processes)
```

Comment:
this is just for test code, so we can simplify this with Linq yes? All with a predicate perhaps?

Reply:
great idea, done